src/parquet/encodings/dictionary-encoding.h

```diff
@@ -194,19 +207,43 @@ class DictEncoderBase {
 
   int hash_table_size() { return hash_table_size_; }
   int dict_encoded_size() { return dict_encoded_size_; }
+  /// Clears all the indices (but leaves the dictionary).
+  void ClearIndices() { buffered_indices_.clear(); }
 
- protected:
-  explicit DictEncoderBase(MemPool* pool, MemoryAllocator* allocator)
-      : hash_table_size_(INITIAL_HASH_TABLE_SIZE),
-        mod_bitmask_(hash_table_size_ - 1),
-        hash_slots_(0, allocator),
-        allocator_(allocator),
-        pool_(pool),
-        dict_encoded_size_(0) {
-    hash_slots_.Assign(hash_table_size_, HASH_SLOT_EMPTY);
-    if (!CpuInfo::initialized()) { CpuInfo::Init(); }
+  /// Encode value. Note that this does not actually write any data, just
+  /// buffers the value's index to be written later.
+  void Put(const T& value);
+
+  std::shared_ptr<Buffer> FlushValues() override {
```

Comment:
Can we combine these to form an `Encode(const T* src, int num_values, OutputStream* dst)` interface to retain the old encoding API ?